### PR TITLE
Add Ideogram4 image-to-image pipeline

### PR DESCRIPTION
**Pull request by lucasruan1618**
# What does this PR do?

This PR introduces support for the **Ideogram4** model in **Img2Img** tasks, expanding the model’s creative capabilities and integration within the Diffusers library.

## Example code

```python
import os
import torch
from diffusers import Ideogram4Img2ImgPipeline, Ideogram4PromptEnhancerHead
from diffusers.utils import load_image


MODEL_ID = "ideogram-ai/ideogram-4-nf4-diffusers"
PROMPT_ENHANCER_HEAD_ID = "diffusers/qwen3-vl-8b-instruct-lm-head"
IMAGE_URL = "https://raw.githubusercontent.com/lucasruan1618/Image_storage/main/Input/cute_cat.png"

prompt_enhancer_head = Ideogram4PromptEnhancerHead.from_pretrained(
    PROMPT_ENHANCER_HEAD_ID,
    torch_dtype=torch.bfloat16,
)

pipe = Ideogram4Img2ImgPipeline.from_pretrained(
    MODEL_ID,
    prompt_enhancer_head=prompt_enhancer_head,
    torch_dtype=torch.bfloat16,
    token=os.environ["HF_TOKEN"],
).to("cuda")

init_image = load_image(IMAGE_URL).convert("RGB")
text_prompt = "wizard dog, Gandalf-inspired, Lord of the Rings aesthetic, majestic yet cute, Studio Ghibli style"

for strength in [0.6, 0.7, 0.8, 0.9, 1.0]:
    image = pipe(
        prompt=text_prompt,
        image=init_image,
        height=init_image.height,
        width=init_image.width,
        strength=strength,
        prompt_upsampling=True,
        generator=torch.Generator(device="cuda").manual_seed(42),
    ).images[0]
    image.save(f"ideogram4_img2img_{strength}.png")
```

## Results at different strengths

The following results were generated after fixing the prompt enhancer temperature to 0, ensuring deterministic prompt enhancement and reproducible outputs across runs with the same seed.

<table>
  <tr>
    <td width="100%"><b>Init image</b><br/>
      
    </td>
  </tr>
</table>

<table>
  <tr>
    <td><b>0.6</b><br/></td>
    <td><b>0.7</b><br/></td>
    <td><b>0.8</b><br/></td>
    <td><b>0.9</b><br/></td>
    <td><b>1.0</b><br/></td>
  </tr>
</table>

## Before submitting

* [x] Did you read the [[contributor guideline](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution)](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution)?
* [x] Did you read our [[philosophy doc](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy)](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy)? (important for complex PRs)
* [ ] Was this discussed/approved via a GitHub issue or the [[forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
* [x] Did you make sure to update the documentation with your changes? Here are the
  [[documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs)](https://github.com/huggingface/diffusers/tree/main/docs), and
  [[here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation)](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
* [x] Did you write any new necessary tests?

## Who can review?

- Pipelines and models: @yiyixuxu @dg845

### CODE REVIEW

**Comment by yiyixuxu** [comment]
Hi @lucasruan1618 
thanks for the PR!
a couple of things before we can review:
(1) can you implement this through modular instead?
(2) can you make sure you've read and respected our [philosophy](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy)? this PR didn't follow it (see the [pipelines section](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy#pipelines)). If you have an AI assistant, we recommend you do a self-review before requesting a review from us — see [here](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#ai-assisted-and-agentic-contributions).

happy to review once these are followed!

**Comment by lucasruan1618** [comment]
Hi @yiyixuxu 

Thanks for the feedback.

Just to make sure I understand correctly: are you suggesting that Ideogram4 Img2Img should be implemented as a Modular Diffusers workflow (reusing the existing Ideogram4 blocks and adding the img2img-specific stages), rather than as a separate `Ideogram4Img2ImgPipeline` under `src/diffusers/pipelines/ideogram4`?

The reason I implemented it as a subclass of the existing `Ideogram4Pipeline` is that the vast majority of the generation workflow is shared with text-to-image (prompt encoding, conditioning, guidance, denoising, scheduler setup, and decoding). The Img2Img implementation only introduces image-specific latent initialization and strength-based timestep handling.

Also, besides the recommendation to use Modular Diffusers, are there any other aspects of the PR that you feel do not align with the Diffusers philosophy or pipeline guidelines? I'd like to address all architectural concerns before reworking the implementation.
